### PR TITLE
feat(app): add ignore-healthcheck annotation

### PR DESCRIPTION
**Pull request by CefBoud**
Closes https://github.com/argoproj/argo-cd/issues/19284

# Description
This PR introduces the `argocd.argoproj.io/ignore-healthcheck` annotation for an Application's immediate children. This allows the health status of these child resources to be disregarded when assessing the overall health of the Application. This is particularly useful when the unhealthiness of a child resource is acceptable and should not affect the Application's health evaluation.

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://githu.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [ ] I have added a brief description of why this PR is necessary and/or what this PR solves.
* [ ] Optional. My organization is added to USERS.md.
* [ ] Optional. For bug fixes, I've indicated what older releases this fix should be cherry-picked into (this may or may not happen depending on risk/complexity).

### CODE REVIEW

**Comment by andrii-korotkov-verkada** [comment]
There may be a better way, but need more context. See comments on the issue.

File: controller/health.go [hunk 1]
```diff
@@ -24,6 +25,9 @@ func setApplicationHealth(resources []managedResource, statuses []appv1.Resource
 		if res.Target != nil && hookutil.Skip(res.Target) {
 			continue
 		}
+		if res.Target != nil && res.Target.GetAnnotations() != nil && res.Target.GetAnnotations()[common.AnnotationIgnoreHealthCheck] == "true" {
```
Comment (andrii-korotkov-verkada) [inline controller/health.go:28]: Are the resources here all app resources or only immediate children?

Comment (CefBoud) [reply controller/health.go:28]: Only immediate childre. As per [documentation](https://argo-cd.readthedocs.io/en/stable/operator-manual/health/#health-checks): `An Argo CD App's health is inferred from the health of its immediate child resources`

**Review by andrii-korotkov-verkada — APPROVED** [review]
Please, rebase and make the tests pass.

**Review by ishitasequeira — APPROVED** [review]
LGTM!!

**Pull request merged**